src/map/status.c

```diff
@@ -10363,7 +10485,7 @@ int status_change_end_(struct block_list* bl, enum sc_type type, int tid, const
 					// Clear Status from others
 					int i;
 					for( i = 0; i < MAX_PC_DEVOTION; i++ ) {
-						if (sd->devotion[i] && (tsd = map->id2sd(sd->devotion[i])) != NULL && tsd->sc.data[type])
+						if (sd && sd->devotion[i] && (tsd = map->id2sd(sd->devotion[i])) != NULL && tsd->sc.data[type])
```

Comment:
This should be moved to outside the for loop. It's, in a way, already guaranteed to be nonnull (because it checks the type of the bl to be `BL_PC`, but a nullpo check would be good documentation.